polaris-react/src/components/IndexFilters/IndexFilters.tsx

```diff
@@ -409,7 +430,7 @@ export function IndexFilters({
                   onQueryFocus={handleQueryFocus}
                   onQueryBlur={handleQueryBlur}
                   onAddFilterClick={onAddFilterClick}
-                  filters={filters}
+                  filters={filters || []}
                   appliedFilters={appliedFilters}
                   onClearAll={onClearAll}
                   disableFilters={disabled}
```

Comment:
Might need adjust the gap at line 445 if the spacing for the loading is incorrect.